server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -544,7 +544,9 @@ public ConnectionManager.ConnectionValidator connectionValidatorForExtensionConn
     ) {
         return (newConnection, actualProfile, listener) -> {
             // We don't validate cluster names to allow for CCS connections.
-            threadPool.getThreadContext().putHeader("extension_unique_id", extensionUniqueId);
+            if (Strings.isNullOrEmpty(threadPool.getThreadContext().getHeader("extension_unique_id"))) {
+                threadPool.getThreadContext().putHeader("extension_unique_id", extensionUniqueId);
```

Comment:
While this works as a fast workaround, I'm wondering if it would be preferable to still throw an exception on a wrong value.  So checking for null or empty OR a value not equal to `extensionUniqueId`.